frontend/ui/lib/main.dart

```diff
@@ -34,6 +37,18 @@ class TripMeOut extends StatelessWidget {
             settings: settings,
           );
         }
+        if (settings.name == '/createTrip') {
+          return MaterialPageRoute(
+            builder: (context) => CreateTripPage(),
+            settings: settings,
+          );
+        }
+        if (settings.name == '/tripView') {
```

Comment:
Would it make sense to call this route `/viewTrip` to match the verb-noun form above? I don't recall how we named the HTML pages earlier.

Reply:
This is gross. The route should be `/trips/${id}` just like the rest path. That means linking to a specific trip page should always give the same user the same page.

To do this you will need to do regex matching/extraction. It's not very clean, but something like this should do the trick:

```dart
RegExp tripViewRouteRegExp = RegExp(r"/trips/(?<tripId>[^/]+)$");
RegExpMatch matchesTripViewRoute = tripViewRouteRegExp.firstMatch(settings.name);
if (matchesTripViewRoute != null) {
  String tripId = matchesTripViewRoute.namedGroup('tripId');
  // use your trip id as a parameter to the page you are about to render.
}
```

Reply:
Ah, that explains why you were using `onGenerateRoute` rather than a `routes` map. Sad to see that this is just about as difficult as in Java servlets. 😞

Reply:
So we will want to change this to do the correct ID thing soon. but I am actually ok with just the view right now.